internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -467,6 +479,114 @@ func (r *BMACReconciler) findBMH(ctx context.Context, agent *adiiov1alpha1.Agent
 	return nil, nil
 }
 
+func (r *BMACReconciler) isDay2Worker(agent *adiiov1alpha1.Agent) bool {
+	return agent.Spec.Role == models.HostRoleWorker && agent.Spec.ClusterDeploymentName != nil
+}
+
+func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost, agent *adiiov1alpha1.Agent) reconcileResult {
+	cdKey := types.NamespacedName{
+		Namespace: agent.Spec.ClusterDeploymentName.Namespace,
+		Name:      agent.Spec.ClusterDeploymentName.Name,
+	}
+	clusterDeployment := &hivev1.ClusterDeployment{}
+	if err := r.Get(ctx, cdKey, clusterDeployment); err != nil {
+		r.Log.WithError(err).Errorf("failed to get clusterDeployment resource %s/%s", cdKey.Namespace, cdKey.Name)
+		return reconcileError{err}
+	}
+
+	// Secret contains kubeconfig for the spoke cluster
+	secret := &corev1.Secret{}
```

Comment:
can we unite logic from here and  from https://github.com/openshift/assisted-service/blob/fd1e93ec8b8614ac358241cac4761fd62128de00/internal/controller/controllers/clusterdeployments_controller.go#L208 
will be nice to have same function for get kubeconfig in both cases